### PR TITLE
- view & update profile 

### PR DESCRIPTION
**Pull request by vthtuy**
- skip authen in API
- fix DB connection message ("connection error TypeError: callback is not a function")

### CODE REVIEW

File: server/routes/profile.js [hunk 1]
```diff
@@ -1,9 +1,22 @@
...
+const profileController = require('../controllers/profile')
 const router = express.Router();
+const User = require('../models/user');
 
 router.get('/check', async (req, res) => {
   res.json({ success: true })
 });
 
+router.get('/:username',   (req, res) => {
+  User.findOne({email : req.params['username']}).then( result => {
```
Comment (vnts7) [inline server/routes/profile.js]: use async/await

File: server/controllers/profile.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const User = require('../models/user');
+ 
+function findByUserName(user) { 
+  // result =  User.findOne({email : user})
+  // console.log('result findOne: ' , result);
+  // return result; 
+  console.log('user: ' , user);
+  User.findOne({email : user}).then( result => {
```
Comment (vnts7) [inline server/controllers/profile.js]: async

File: server/models/user.js [hunk 3]
```diff
@@ -1,32 +1,79 @@
...
-    required: true
-  },
-  email: {
-    type: String,
-    required: true,
-    unique: true,
-    // Regexp to validate emails with more strict rules as added in tests/users.js which also conforms mostly with RFC2822 guide lines
-    match: [/^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/, 'Please enter a valid email'],
-  },
-  password: {
```
Comment (vnts7) [inline server/models/user.js]: why remove password?

File: server/models/user.js [hunk 4]
```diff
@@ -1,32 +1,79 @@
...
-  },
-  createdAt: {
-    type: Date,
-    default: Date.now
-  },
-  roles: [{
-    type: String,
-  }]
-}, {
+    username: {
```
Comment (vnts7) [inline server/models/user.js]: pls use 2 space tab size

File: src/app/profile/profile.component.ts [hunk 5]
```diff
@@ -1,15 +1,28 @@
...
 @Component({
   selector: 'app-profile',
   templateUrl: './profile.component.html',
   styleUrls: ['./profile.component.css']
 })
 export class ProfileComponent implements OnInit {
+  
+  sub : Subscription;
+  
+  user={email: 'hmm@gmail.com', name: "Handsome Mohammed"}
```
Comment (vnts7) [inline src/app/profile/profile.component.ts]: get  user from AuthService

File: server/models/user.js (hunk 4 above)
Comment (vnts7) [inline server/models/user.js]: remove username

File: src/app/profile/profile.component.html [hunk 6]
```diff
@@ -1,3 +1,28 @@
-<p>
-  profile works!
-</p>
+<div class="c-box">
```
Comment (vnts7) [inline src/app/profile/profile.component.html]: WHy dont use edit form from begining?

File: server/routes/profile.js (hunk 1 above)
Comment (vthtuy) [reply server/routes/profile.js]: done https://github.com/vnts7/mwa-prj/pull/10/commits/4b53bf8ff5ab0c7eb7e0bcc2dc4aa0a0bd83ab65

File: server/controllers/profile.js (hunk 2 above)
Comment (vthtuy) [reply server/controllers/profile.js]: done https://github.com/vnts7/mwa-prj/pull/10/commits/4b53bf8ff5ab0c7eb7e0bcc2dc4aa0a0bd83ab65

File: server/models/user.js (hunk 3 above)
Comment (vthtuy) [reply server/models/user.js]: done https://github.com/vnts7/mwa-prj/pull/10/commits/f0ee6bc5c2271b896d1cf6daf5586c73a54b01e7

File: server/models/user.js (hunk 4 above)
Comment (vthtuy) [reply server/models/user.js]: done https://github.com/vnts7/mwa-prj/pull/10/commits/f769d508e1193f20d3875614c53535b3a49e3d97

Comment (vthtuy) [reply server/models/user.js]: done

File: src/app/profile/profile.component.ts (hunk 5 above)
Comment (vthtuy) [reply src/app/profile/profile.component.ts]: done https://github.com/vnts7/mwa-prj/pull/10/commits/a690ea8fde8a6d56391923253372233665f12ed0

File: src/app/profile/profile.component.html (hunk 6 above)
Comment (vthtuy) [reply src/app/profile/profile.component.html]: 2 parts

File: server/controllers/profile.controller.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+const User = require('../models/user');
+ 
+async function findByUserName(user) {  
+  console.log('user: ' , user);
+  await User.findOne({email : user}).then( result => {
```
Comment (vnts7) [inline server/controllers/profile.controller.js]: .then?

File: server/routes/profile.js [hunk 8]
```diff
@@ -1,9 +1,24 @@
...
+const profileController = require('../controllers/profile.controller')
 const router = express.Router();
+const User = require('../models/user');
 
 router.get('/check', async (req, res) => {
   res.json({ success: true })
 });
 
+router.get('/:username', async  (req, res) => {  
+  let data = await profileController.findByUserName(req.params['username']);
```
Comment (vnts7) [inline server/routes/profile.js]: use req.user._id

File: src/app/login/login.component.html [hunk 9]
```diff
@@ -11,7 +11,7 @@ <h1>Login</h1>
       <input matInput placeholder="Password" type="password" name="password" formControlName="password" required>
       <mat-icon matSuffix>lock</mat-icon>
     </mat-form-field>
-    <button mat-raised-button color="warn" type="submit">Login</button>
+    <button mat-raised-button color="primary" type="submit">Login</button>
```
Comment (vnts7) [inline src/app/login/login.component.html]: why?

File: src/app/profile/profile.component.html [hunk 10]
```diff
@@ -1,3 +1,81 @@
...
+              </mat-form-field>
+              <mat-form-field>
+                <mat-label>Gender</mat-label> 
+                <mat-select formControlName="gender" >
+                  <mat-option value="0">Male</mat-option>
+                  <mat-option value="1">Female</mat-option>
+                </mat-select>
+              </mat-form-field>
+              <mat-form-field>
+                <input matInput placeholder="Date Of Birth" formControlName="dob" type="date">
```
Comment (vnts7) [inline src/app/profile/profile.component.html]: use matDatePicker

File: src/app/profile/profile.component.ts [hunk 11]
```diff
@@ -1,15 +1,56 @@
...
+  }
+
+  onSave(event) {
+      this.message = null; 
+      const formData = this.form.getRawValue(); 
+      this.sub = this.profileService.updateProfile(formData).subscribe( rs => {     
+        console.log('onSave', rs) 
+        // this.message = JSON.parse(rs).message;
+        this.selectedIndex = 1;
+        this.loadData();
```
Comment (vnts7) [inline src/app/profile/profile.component.ts]: use this.user = rs.data;

File: server/controllers/profile.controller.js [hunk 12]
```diff
@@ -0,0 +1,11 @@
+const User = require('../models/user');
+ 
+async function findByUserName(user) {
+  return await User.findOne({email : user});
+}
+
+async function updateProfile(user) {
+    return await User.updateOne({ _id: user._id }, user);
```
Comment (vnts7) [inline server/controllers/profile.controller.js]: where is caloriesNeeds calculate?

File: server/routes/profile.js (hunk 8 above)
Comment (vthtuy) [reply server/routes/profile.js]: Got the issue when using req.user._id 
[SERVER] (node:55564) UnhandledPromiseRejectionWarning: TypeError: Cannot read property '_id' of undefined

I think it does not matter with this API.

Comment (vthtuy) [reply server/routes/profile.js]: I will keep it as it is

File: src/app/login/login.component.html (hunk 9 above)
Comment (vthtuy) [reply src/app/login/login.component.html]: warn is in RED color for warning alert. We should keep Login as primary, it matches with the current theme. It's better.

File: src/app/profile/profile.component.html (hunk 10 above)
Comment (vthtuy) [reply src/app/profile/profile.component.html]: input="date" support date formate restriction when user puts random date. It's a good choice.

will keep it as it is.

File: src/app/profile/profile.component.ts (hunk 11 above)
Comment (vthtuy) [reply src/app/profile/profile.component.ts]: somehow rs is null.
And this is after updating data. We need to retrieve data again.
Will keep it as it is

File: server/controllers/profile.controller.js (hunk 12 above)
Comment (vthtuy) [reply server/controllers/profile.controller.js]: done https://github.com/vnts7/mwa-prj/pull/10/commits/e603276aa1a232976b2e4fd50d1075283c85a49c

**Pull request merged**